hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/execution/benchmark/SpaceCurveOptimizeBenchMark.scala

```diff
@@ -0,0 +1,132 @@
+        getSkippingPercent("table_hilbert_sort_byMap", "c1_int", "c2_int", 500000, 500000)
+        getSkippingPercent("table_hilbert_sort_bySample", "c1_int", "c2_int", 500000, 500000)
+      }
+    }
+  }
+
+  /*
+  for table table_z_sort_byMap_skew with query filter: c1_int = 5000 or c2_int = 500000 we can achieve skipping percent 0.0
+  for table table_z_sort_bySample_skew with query filter: c1_int = 5000 or c2_int = 500000 we can achieve skipping percent 0.78
+  for table table_hilbert_sort_byMap_skew with query filter: c1_int = 5000 or c2_int = 500000 we can achieve skipping percent 0.05500000000000005
+  for table table_hilbert_sort_bySample_skew with query filter: c1_int = 5000 or c2_int = 500000 we can achieve skipping percent 0.84
+  */
+  def runSkewTableSkippingBenchMark(): Unit = {
+    withTempDir { f =>
+      withTempTable("table_z_sort_byMap_skew", "table_z_sort_bySample_skew", "table_hilbert_sort_byMap_skew", "table_hilbert_sort_bySample_skew") {
+        prepareSkewTable(new Path(f.getAbsolutePath), 1000000)
+        // choose median value as filter condition.
+        // the median value of c1_int is 5000
+        // the median value of c2_int is 500000
+        getSkippingPercent("table_z_sort_byMap_skew", "c1_int", "c2_int", 5000, 500000)
+        getSkippingPercent("table_z_sort_bySample_skew", "c1_int", "c2_int", 5000, 500000)
+        getSkippingPercent("table_hilbert_sort_byMap_skew", "c1_int", "c2_int", 5000, 500000)
+        getSkippingPercent("table_hilbert_sort_bySample_skew", "c1_int", "c2_int", 5000, 500000)
+      }
+    }
+  }
+
+  def main(args: Array[String]): Unit = {
+    runNormalTableSkippingBenchMark()
+    runSkewTableSkippingBenchMark()
+  }
+
+  def withTempTable(tableNames: String*)(f: => Unit): Unit = {
+    try f finally tableNames.foreach(spark.catalog.dropTempView)
+  }
+
+  def prepareNormalTable(tablePath: Path, numRows: Int): Unit = {
+    import spark.implicits._
+    val df = spark.range(numRows).map(_ => (Random.nextInt(1000000), Random.nextInt(1000000))).toDF("c1_int", "c2_int")
+    val dfOptimizeByMap = SpaceCurveOptimizeHelper.createOptimizedDataFrameByMapValue(df, "c1_int, c2_int", 200, "z-order")
+    val dfOptimizeBySample = SpaceCurveOptimizeHelper.createOptimizeDataFrameBySample(df, "c1_int, c2_int", 200, "z-order")
+
+    val dfHilbertOptimizeByMap = SpaceCurveOptimizeHelper.createOptimizedDataFrameByMapValue(df, "c1_int, c2_int", 200, "hilbert")
+    val dfHilbertOptimizeBySample = SpaceCurveOptimizeHelper.createOptimizeDataFrameBySample(df, "c1_int, c2_int", 200, "hilbert")
+
+    saveAsTable(dfOptimizeByMap, tablePath, "z_sort_byMap")
+    saveAsTable(dfOptimizeBySample, tablePath, "z_sort_bySample")
+    saveAsTable(dfHilbertOptimizeByMap, tablePath, "hilbert_sort_byMap")
+    saveAsTable(dfHilbertOptimizeBySample, tablePath, "hilbert_sort_bySample")
+  }
+
+  def prepareSkewTable(tablePath: Path, numRows: Int): Unit = {
```

Comment:
No need to duplicate this, instead please parameterize the other method to either accept desired range of the random point axis, or generator producing random point as a whole

Reply:
ok， will do it。
but
this class is a benchmark class， not a UT test。 This is just a sample class showing performance improvement

Reply:
fixed it